### DISCUSSION TITLE
Max Voltage Setpoint changes

### DISCUSSION THREAD

**Discussion by majic79** [Louisvdw/dbus-serialbattery#855] [Q&A]
Going to keep this simple - what influences the max_voltage setpoint for charging?

![Charge Curve w VMax](https://github.com/Louisvdw/dbus-serialbattery/assets/1041562/6be4a6ad-339b-4ec2-b0c2-b404cfebdfb6)

I'm charging to 3.55V cell (balancing in JKBMS set to 3.55V) and I really need it to stay there a bit longer than 10 minutes to give it a chance to work. The setpoint is all over the place at the end of bulk charging, then it goes into absorbtion (seen by the charger indicating the change in state) and the max voltage setpoint is everywhere. That's less of an issue as it resolves soon after, and holds at the max voltage for charge (3.55V x16 = 56.8v) for a few minutes then it drops to the float voltage (3.45V x16 = 55.2V)

My last change 2 days ago was to set the "MAX_VOLTAGE_TIME_SEC" to 7200 (2 hours) but that seems to have had no impact on things. I was hoping that the max voltage time factor would only kick in after it's gone to Absorption (current drops below some threshold) but it doesn't appear to work that way. Is there a way to inhibit the flot voltage change entirely? I want the linear current regulation, but I don't want it to drop to the Float voltage until much later

**Comment by majic79** [comment]
Config file below:

```
[DEFAULT]

BMS_TYPE = Jkbms
BATTERY_CAPACITY = 230

MAX_BATTERY_CHARGE_CURRENT = 115.0
MAX_BATTERY_DISCHARGE_CURRENT = 115.0
MAX_VOLTAGE_TIME_SEC = 7200
LINEAR_LIMITATION_ENABLE = True
CCCM_CV_ENABLE = True
DCCM_CV_ENABLE = True
CCCM_T_ENABLE = True
DCCM_T_ENABLE = True
CCCM_SOC_ENABLE = False
DCCM_SOC_ENABLE = False

CELL_VOLTAGES_WHILE_CHARGING   = 3.65, 3.60, 3.55, 3.40
MAX_CHARGE_CURRENT_CV_FRACTION = 0.02, 0.50, 1.00, 1.00
CELL_VOLTAGES_WHILE_DISCHARGING   = 2.60, 2.70, 2.80, 3.0
MAX_DISCHARGE_CURRENT_CV_FRACTION =    0,  0.1,  0.5,   1
CVCM_ENABLE = True

MIN_CELL_VOLTAGE   = 2.750
MAX_CELL_VOLTAGE   = 3.550
FLOAT_CELL_VOLTAGE = 3.450

TEMPERATURE_LIMITS_WHILE_CHARGING = 0,  10, 40, 55
MAX_CHARGE_CURRENT_T_FRACTION     = 0,   1,  1,  0

TEMPERATURE_LIMITS_WHILE_DISCHARGING = -20,    0,   10, 20, 45, 55
MAX_DISCHARGE_CURRENT_T_FRACTION     =   0, 0.95, 0.97,  1,  1,  0

````

**Reply by majic79** [reply]
Battery is 16x LiFePO4 CALB cells (230AH each), Charger is a Victron MultiPlus-II 48/5000/70-50, HomeAssistant integration over MQTT for data collection

**Reply by majic79** [reply]
do I need CVCM for LFP cells?

**Comment by mr-manuel** [comment]
It's hard to help people when they open a discussion, but the discussion is more a support request then a discussion and a great part of needed info is missing.

Try the latest `nightly` from the `dev` branch. If that does not help open a support request under issues.

**Reply by majic79** [reply]
That's why I opened it as "Q&A". I don't know if this is an issue (happy to raise the ticket) or if there's some other factor (hence "Questions and Answers" and the question was at the top of the first post)

Not interested in jumping to a nightly at this stage, I want to know more before I go that route. I'll open up a new discussion (Q&A) about CVCM, and open up a ticket on the suspected issue with MAX_VOLTAGE_TIME_SEC parameter being ignored

**Reply by mr-manuel** [reply]
Uncomment line 243 and comment line 254. This is already changed in the nightly since a few months.

https://github.com/Louisvdw/dbus-serialbattery/blob/368174813bbdbc091f644f9e8fa401e8fd4cfe9f/etc/dbus-serialbattery/battery.py#L243-L245
